zerver/lib/create_user.py

```diff
@@ -127,4 +129,21 @@ def create_user(email: str, password: Optional[str], realm: Realm,
     user_profile.save(update_fields=["recipient"])
 
     Subscription.objects.create(user_profile=user_profile, recipient=recipient)
+
+    if avatar_source == UserProfile.AVATAR_FROM_SOCIAL_AUTH:
+        from zerver.lib.upload import upload_avatar_image_from_url
+        from zerver.lib.avatar_hash import user_avatar_content_hash
+        from zerver.lib.actions import do_change_avatar_fields
+        from zerver.lib.actions import avatar_url
+        from django.core.files import File
+
+        file_path = upload_avatar_image_from_url(user_avatar_url, user_profile)
+        do_change_avatar_fields(user_profile, UserProfile.AVATAR_FROM_USER)
+        user_avatar_url = avatar_url(user_profile)
+        file_path = "/srv/zulip/var/uploads/avatars/" + file_path + ".png"
+        avatar_file = open(file_path, 'rb')
+        django_avatar_file = File(avatar_file)
+        user_profile.avatar_hash = user_avatar_content_hash(django_avatar_file.read())
+        user_profile.save(update_fields=["avatar_hash"])
```

Comment:
If you make `upload_avatar_image_from_url` return the hash, this block should become much easier. And  computing the hash will be much easier for that function too, as it already has the image content.

Reply:
Yeees, code got much cleaner :)

Reply:
So, one thing I'm concerned about with this model is we're taking what was previously a path that could never fail and adding the possibility of download/upload failures here.  We will likely at least want a try/except block to make sure the code after this runs even if importing from GitHub fails.